Source/WebCore/dom/Document.cpp

```diff
@@ -9820,8 +9820,14 @@ CompositeOperator Document::compositeOperatorForBackgroundColor(const Color& col
 void Document::didAssociateFormControl(Element& element)
 {
     RefPtr page = this->page();
-    if (!page || !page->chrome().client().shouldNotifyOnFormChanges())
+    if (!page)
         return;
+    if (!page->chrome().client().shouldNotifyOnFormChanges()) {
+        // Is it sufficient to only support registering event listeners on the window object, or do we need to support other event targets?
+        // Is this check fast enough to prevent perf slowdown?
+        if (m_domWindow && !m_domWindow->hasEventListeners(eventNames().webkitassociateformcontrolsEvent))
```

Comment:
You wanna check `document.hasEventListenersOfType` instead. That map will contain the event listeners throughout the document, not just the ones registered at m_domWindow.